frappe/utils/global_search.py

```diff
@@ -373,11 +373,42 @@ def sync_global_search():
 	:param flags:
 	:return:
 	"""
-	while frappe.cache.llen("global_search_queue") > 0:
-		# rpop to follow FIFO
-		# Last one should override all previous contents of same document
-		value = json.loads(frappe.cache.rpop("global_search_queue").decode("utf-8"))
-		sync_value(value)
+	from itertools import islice
+
+	def get_search_queue_items_generator():
+		while frappe.cache.llen("global_search_queue") > 0:
+			item_json = frappe.cache.rpop("global_search_queue").decode("utf-8")
+			item_values = json.loads(item_json).values()
+			yield tuple(item_values)
+
+	search_queue_items = get_search_queue_items_generator()
+
+	while values := tuple(islice(search_queue_items, 10_000)):
```

Comment:
There's create batch utils somewhere in framework.

Also can we somehow pop from redis in batches too? 🤔 

(Need to ensure that we don't miss anything though)

Reply:
Another question: what happens when same document gets modified twice and is sent twice via same query. DBs might not like it 😄

Reply:
`RPOP KEY LEN` can performed, will have to alter the wrapper though, should I do that?

Reply:
Okay, nvm the `LEN` argument was added in 6.20 we (CI, FC) run an older version of redis which doesn't have that.

Reply:
Also `utils.create_batch` won't work for this purpose cause while it asks for an `Iterable`, before yielding it slices the `Iterable`.

`Iterable` does not imply sliceable.

Misconstrued typing aside, all items will have to popped from Redis for `utils.create_batch`'s use to be justified.